artifactory/commands/transferfiles/manager.go

```diff
@@ -188,30 +189,41 @@ func (ptm *PollingTasksManager) stop() {
 }
 
 type producerConsumerDetails struct {
-	producerConsumer parallel.Runner
-	errorsQueue      *clientUtils.ErrorsQueue
+	chunkBuilderProducerConsumer  parallel.Runner
+	chunkUploaderProducerConsumer parallel.Runner
+	errorsQueue                   *clientUtils.ErrorsQueue
 }
 
 func initProducerConsumer() producerConsumerDetails {
-	producerConsumer := parallel.NewRunner(GetThreads(), tasksMaxCapacity, false)
+	chunkUploaderProducerConsumer := parallel.NewRunner(GetThreads(), tasksMaxCapacity, false)
+	chunkBuilderProducerConsumer := parallel.NewRunner(GetThreads(), tasksMaxCapacity, false)
 	errorsQueue := clientUtils.NewErrorsQueue(1)
 
 	return producerConsumerDetails{
-		producerConsumer: producerConsumer,
-		errorsQueue:      errorsQueue,
+		chunkUploaderProducerConsumer: chunkUploaderProducerConsumer,
+		chunkBuilderProducerConsumer:  chunkBuilderProducerConsumer,
+		errorsQueue:                   errorsQueue,
 	}
 }
 
 func runProducerConsumer(pcDetails producerConsumerDetails, runWaitGroup *sync.WaitGroup) (runnerErr error, executionErr error) {
-	runWaitGroup.Add(1)
+	runWaitGroup.Add(2)
+	// When the producer consumer is idle for assumeProducerConsumerDoneWhenIdleForSeconds (not tasks are being handled)
+	// the work is assumed to be done.
```

Comment:
...not tasks are being... --> ...no tasks are being...

Reply:
done